### PR TITLE
Semi-Final Working Obstacle Avoidance

### PR DESCRIPTION
**Pull request by BrightStraightReality**
Pls

### CODE REVIEW

File: SDAPackage/SDA/drone.py [hunk 1]
```diff
@@ -23,6 +23,10 @@ def set_drone_position(self, new_point):
         :param new_point: New point for the drone
         :type new_point: Numpy Array
         """
+        print('current point of drone')
+        print(self.point)
+        print('new point of drone')
+        print(new_point)
```
Comment (DarkmatterVale) [inline SDAPackage/SDA/drone.py]: All of these comments should be removed.

File: SDAPackage/SDA/obstacle_map.py [hunk 2]
```diff
@@ -133,6 +133,7 @@ def is_obstacle_in_path(self):
                 if isinstance(obstacle, StationaryObstacle):
                     paths = self.generate_possible_paths(obstacle)
                     if len(paths) != 0:
+                        print(len(paths))
```
Comment (DarkmatterVale) [inline SDAPackage/SDA/obstacle_map.py]: Comment should be removed.

File: SDAPackage/SDA/obstacle_map.py [hunk 3]
```diff
@@ -158,20 +159,24 @@ def generate_possible_paths(self, obstacle):
                     [obstacle.get_point()[0] + obstacle.get_radius(), obstacle.get_point()[1], obstacle.get_height() + (Constants.STATIONARY_OBSTACLE_SAFETY_RADIUS)],
                     [obstacle.get_point()[0] - obstacle.get_radius(), obstacle.get_point()[1], obstacle.get_height() + (Constants.STATIONARY_OBSTACLE_SAFETY_RADIUS)]
                 ]
+                #for new_point in new_attempt_pos_points:
+                #    print("New Attempt Pos Point", str(new_point))
```
Comment (DarkmatterVale) [inline SDAPackage/SDA/obstacle_map.py]: Remove these commented out debug statements.

File: SDAPackage/SDA/obstacle_map.py [hunk 4]
```diff
@@ -133,6 +133,7 @@ def is_obstacle_in_path(self):
                 if isinstance(obstacle, StationaryObstacle):
                     paths = self.generate_possible_paths(obstacle)
                     if len(paths) != 0:
+                        print(len(paths))
                         return True, np.array(paths)
                 #elif isinstance(obstacle, MovingObstacle):
                 #    pass
```
Comment (DarkmatterVale) [inline SDAPackage/SDA/obstacle_map.py]: We have removed moving obstacles from this iteration of SDA, so please remove the commented out code here.

File: SDAPackage/SDA/obstacle_map.py [hunk 5]
```diff
@@ -158,20 +159,24 @@ def generate_possible_paths(self, obstacle):
...
                 ]
+                #for new_point in new_attempt_pos_points:
+                #    print("New Attempt Pos Point", str(new_point))
 
                 new_paths = []
                 for new_pos_point in new_attempt_pos_points:
-                    if not self.does_path_intersect_obstacle_3d(obstacle, self.drone.get_point(), new_pos_point) and self.flight_boundary.is_point_in_bounds(new_pos_point):
+                    print(self.flight_boundary.is_point_in_bounds(new_pos_point))
+                    if not self.does_path_intersect_obstacle_2d(obstacle, self.drone.get_point(), new_pos_point) and self.flight_boundary.is_point_in_bounds(new_pos_point):
+                    #if self.flight_boundary.is_point_in_bounds(new_pos_point):
```
Comment (DarkmatterVale) [inline SDAPackage/SDA/obstacle_map.py]: Remove the commented out code and the print statement.

File: SDAPackage/SDA/obstacle_map.py [hunk 6]
```diff
@@ -158,20 +159,24 @@ def generate_possible_paths(self, obstacle):
...
                             if self.flight_boundary.is_point_in_bounds(recursive_new_pos_point) and abs(recursive_new_pos_point[2] - new_pos_point[2]) < 5:
                                 if recursive_new_pos_point[0] != new_pos_point[0] or recursive_new_pos_point[1] != new_pos_point[1]:
-                                    if not self.does_path_intersect_obstacle_3d(obstacle, new_pos_point, recursive_new_pos_point) and not self.does_path_intersect_obstacle_3d(obstacle, recursive_new_pos_point, self.drone.get_waypoint_holder().get_current_waypoint()):
+                                    if not self.does_path_intersect_obstacle_2d(obstacle, new_pos_point, recursive_new_pos_point) and not self.does_path_intersect_obstacle_2d(obstacle, recursive_new_pos_point, self.drone.get_waypoint_holder().get_current_waypoint()):
                                         new_paths.append([new_pos_point, recursive_new_pos_point])
 
                 # Uncomment for DEBUGGING ONLY
                 for path in new_paths:
+                    print(self.get_path_distance(path))
                     print("Point:", str(path))
```
Comment (DarkmatterVale) [inline SDAPackage/SDA/obstacle_map.py]: I like the additional print statement. Since these comments are only for debugging, please comment them all out.

File: SDAPackage/SDA/obstacle_map.py [hunk 7]
```diff
@@ -158,20 +159,24 @@ def generate_possible_paths(self, obstacle):
...
-
```
Comment (DarkmatterVale) [inline SDAPackage/SDA/obstacle_map.py]: Please add back in this new line.

File: SDAPackage/SDA/obstacle_map.py [hunk 8]
```diff
@@ -292,7 +266,8 @@ def get_min_path(self, paths):
             if distance < shortest_distance:
                 shortest_path = path
                 shortest_distance = distance
-
+        print('min path')
+        print(shortest_path)
```
Comment (DarkmatterVale) [inline SDAPackage/SDA/obstacle_map.py]: Please delete these print statements.

File: SUASSystem/SUASSystem/gcs.py [hunk 9]
```diff
@@ -23,13 +24,15 @@ def gcs_process(sda_status, img_proc_status, interop_client_array, targets_to_su
...
     sda_avoid_coords = manager.list()
+    UAV_status = manager.Value('s', "AUTO")
     location_log = manager.list()
     vehicle = connect_to_vehicle()
     waypoints = download_waypoints(vehicle)
 
     if len(interop_client_array) != 0:
         mission_information_data.append(get_mission_json(interop_client_array[0].get_active_mission(), interop_client_array[0].get_obstacles()))
     else:
+        print('mission information is empty')
```
Comment (DarkmatterVale) [inline SUASSystem/SUASSystem/gcs.py]: Lets make this print statement a little more professional please. Something along the lines of:

```print("[Error] : The GCS process is unable to load mission data from the Interoperability server")```

File: SUASSystem/SUASSystem/gcs.py [hunk 10]
```diff
@@ -48,24 +50,25 @@ def gcs_process(sda_status, img_proc_status, interop_client_array, targets_to_su
...
             mission_information_data[0] = get_mission_json(interop_client_array[0].get_active_mission(), interop_client_array[0].get_obstacles())
 
-        if sda_status.value.lower() == "connected":
-            """if (vehicle.location.global_relative_frame.alt * 3.28084) > GCSSettings.SDA_MIN_ALT and (vehicle.mode.name == "GUIDED" or vehicle.mode.name == "AUTO"):
+        if (vehicle.location.global_relative_frame.alt * 3.28084) > GCSSettings.SDA_MIN_ALT:#and sda_status.value.lower() == "connected":
+            print("Condition is TRUE")
+            if (UAV_status.value == "GUIDED"):
+                sda_avoid_feet_height = Location(sda_avoid_coords[0].get_lat(), sda_avoid_coords[0].get_lon(), sda_avoid_coords[0].get_alt()*3.28084)
+                print('sda avoid feet height')
+                print(sda_avoid_feet_height)
```
Comment (DarkmatterVale) [inline SUASSystem/SUASSystem/gcs.py]: Lets remove these print statements.

File: SUASSystem/SUASSystem/gcs.py [hunk 11]
```diff
@@ -48,24 +50,25 @@ def gcs_process(sda_status, img_proc_status, interop_client_array, targets_to_su
...
                 log("root", "Avoiding obstacles...")
-                vehicle.mode = VehicleMode("GUIDED")
-                guided_waypoint_location = sda_avoid_coords[0][0]
-                vehicle.simple_goto(guided_waypoint_location.as_global_relative_frame())
+                vehicle.mode = dronekit.VehicleMode("GUIDED")
+                vehicle.simple_goto(sda_avoid_coords[0].as_global_relative_frame())
+            if UAV_status.value == 'AUTO' and vehicle.mode.name != "AUTO":
+                vehicle.mode = dronekit.VehicleMode("AUTO")
```
Comment (DarkmatterVale) [inline SUASSystem/SUASSystem/gcs.py:67]: All of this needs to be encapsulated in the ```if sda_status.value.lower() == "connected":```

File: SUASSystem/SUASSystem/sda.py [hunk 12]
```diff
@@ -2,44 +2,94 @@
...
+    '''boundary_points = [
+            {
+                "boundary_pts" : [
+                    {"latitude" : 38.1424887, "longitude" : -76.4344168, "order" : 0},
+                    {"latitude" : 38.1430962, "longitude" : -76.4248037, "order" : 1},
+                    {"latitude" : 38.1517193, "longitude" : -76.4261985, "order" : 2},
+                    {"latitude" : 38.1513987, "longitude" : -76.4351463, "order" : 3}
+                ]
+            }
+        ]'''
```
Comment (DarkmatterVale) [inline SUASSystem/SUASSystem/sda.py]: Please remove this.

File: SUASSystem/SUASSystem/sda.py [hunk 13]
```diff
@@ -2,44 +2,94 @@
...
     log(logger_name, "Instantiating SDA converter")
     while True:
         try:
+            print('tries to instantiate')
+            #sda_converter = SUASSystem.SDAConverter(vehicle_state_data[0].get_location(), 5)#mission_information_data[0]["fly_zones"])
```
Comment (DarkmatterVale) [inline SUASSystem/SUASSystem/sda.py]: Please remove unnecessary code and the print statement.

File: SUASSystem/SUASSystem/sda.py [hunk 14]
```diff
@@ -2,44 +2,94 @@
...
-
-            sda_converter.set_uav_position(current_location)
-            sda_converter.avoid_obstacles()
-
-            if not sda_converter.has_uav_completed_guided_path():
-                try:
-                    sda_avoid_coords[0] = sda_converter.get_uav_avoid_coordinates()
-                except:
-                    sda_avoid_coords.append(sda_converter.get_uav_avoid_coordinates())
+        if True:#sda_status.value == "connected":
```
Comment (DarkmatterVale) [inline SUASSystem/SUASSystem/sda.py]: Connection check needs to be enabled.

File: SUASSystem/SUASSystem/sda.py [hunk 15]
```diff
@@ -2,44 +2,94 @@
...
+                current_location = vehicle_state_data[0].get_location()
+                current_location.alt = current_location.get_alt() - SUASSystem.GCSSettings.MSL_ALT
+                current_waypoint_number = vehicle_state_data[0].get_current_waypoint_number()
+
+                if current_waypoint_number != 0:
+                    current_uav_waypoint = waypoints[current_waypoint_number - 1]
+                    current_uav_waypoint_location = SUASSystem.Location(current_uav_waypoint.x, current_uav_waypoint.y, current_uav_waypoint.z * 3.28084)
+
+                    print(current_uav_waypoint)
+                    print("IS WAYPOINT AVOIDABLE: " + str(is_waypoint_avoidable(current_uav_waypoint)))
```
Comment (DarkmatterVale) [inline SUASSystem/SUASSystem/sda.py]: Remove these print statements.

File: SUASSystem/SUASSystem/sda.py [hunk 16]
```diff
@@ -2,44 +2,94 @@
...
+                    if is_waypoint_avoidable(current_uav_waypoint) and not is_uav_at_location(current_location, current_uav_waypoint_location):
+                        sda_converter.set_waypoint(current_uav_waypoint_location)
+
+                        #sda_converter.reset_obstacles()
+                        for stationary_obstacle in mission_information_data[0]["stationary_obstacles"]:
+                            sda_converter.add_obstacle(stationary_obstacle)
+                        #for moving_obstacle in mission_information_data["moving_obstacles"]:
+                        #    sda_converter.add_obstacle(get_obstacle_location(moving_obstacle, MSL_ALT), moving_obstacle)
```
Comment (DarkmatterVale) [inline SUASSystem/SUASSystem/sda.py]: Since we are not doing moving obstacles in this iteration of SDA anymore, lets remove this code.

File: SUASSystem/SUASSystem/sda.py [hunk 17]
```diff
@@ -2,44 +2,94 @@
...
+                        sda_converter.set_uav_position(current_location)
+                        if not UAV_status.value == "GUIDED":
+                            sda_converter.avoid_obstacles()
+                        print('current xy coords')
+                        print(SUASSystem.convert_to_point(starting_coords, current_location))
+                        print('current sda avoid coords')
+                        print(sda_avoid_coords)
+                        print("has completed guided path?")
+                        print(sda_converter.has_uav_completed_guided_path())
```
Comment (DarkmatterVale) [inline SUASSystem/SUASSystem/sda.py]: Clean up these print statements. We shouldn't have any unnecessary prints for production, and those that are necessary should look professional and take up as little space as possible.

File: SUASSystem/SUASSystem/sda.py [hunk 18]
```diff
@@ -2,44 +2,94 @@
...
+                        if sda_converter.has_uav_completed_guided_path():
+                            UAV_status.value = "AUTO"
+                            sda_converter.current_path = numpy.array([])
+            except:
+                traceback.print_exc()
 
         sleep(0.5)
+
+def is_waypoint_avoidable(uav_waypoint):
+    non_avoidable_waypoint_command_types = [22, 183, 20]
```
Comment (DarkmatterVale) [inline SUASSystem/SUASSystem/sda.py:62]: Perhaps this list should be placed into the ```GCSSettings``` class?

File: SUASSystem/SUASSystem/sda_converter.py [hunk 19]
```diff
@@ -21,8 +21,7 @@ def __init__(self, initial_coordinates, fly_zones):
         self.initial_coordinates = initial_coordinates
         self.current_path = numpy.array([])
         self.current_path_index = 1
-        self.minimum_change_in_guided_waypoint = 3
-
+        self.minimum_change_in_guided_waypoint = 1000
```
Comment (DarkmatterVale) [inline SUASSystem/SUASSystem/sda_converter.py]: This is not correct. Please reduce this.

File: SUASSystem/SUASSystem/sda_converter.py [hunk 20]
```diff
@@ -61,11 +60,12 @@ def set_waypoint(self, new_waypoint):
         :type new_waypoint: Location
         """
         converted_waypoint = convert_to_point(self.initial_coordinates, new_waypoint)
-
+        print('waypoint')
+        print(converted_waypoint)
```
Comment (DarkmatterVale) [inline SUASSystem/SUASSystem/sda_converter.py]: Remove print statements.

File: SUASSystem/SUASSystem/sda_converter.py [hunk 21]
```diff
@@ -74,11 +74,20 @@ def add_obstacle(self, obstacle_location, obstacle):
...
+
+        obstacle_location = Location(obstacle["latitude"], obstacle["longitude"], 0)
         converted_obstacle_location = convert_to_point(self.initial_coordinates, obstacle_location)
+        print("obstacle")
+        print(converted_obstacle_location)
+        new_obstacle = StationaryObstacle(converted_obstacle_location, obstacle["cylinder_radius"], obstacle["cylinder_height"])
+        #new_obstacle = StationaryObstacle(obstacle.get_point(), obstacle.cylinder_radius, obstacle.cylinder_height)
+        print('ADDS THE OBSTACLE')
+        self.obstacle_map.add_obstacle(new_obstacle)
+        #converted_obstacle_location = convert_to_point(self.initial_coordinates, obstacle_location)
```
Comment (DarkmatterVale) [inline SUASSystem/SUASSystem/sda_converter.py]: Please clean this method up.

File: SUASSystem/SUASSystem/sda_converter.py [hunk 22]
```diff
@@ -74,11 +74,20 @@ def add_obstacle(self, obstacle_location, obstacle):
...
 
-        new_obstacle = StationaryObstacle(converted_obstacle_location, obstacle.cylinder_radius, obstacle.cylinder_height)
+        #new_obstacle = StationaryObstacle(converted_obstacle_location, obstacle.cylinder_radius, obstacle.cylinder_height)
 
-        self.obstacle_map.add_obstacle(new_obstacle)
+        #self.obstacle_map.add_obstacle(new_obstacle)
```
Comment (DarkmatterVale) [inline SUASSystem/SUASSystem/sda_converter.py]: Should this be commented out?

File: SUASSystem/SUASSystem/sda_converter.py [hunk 23]
```diff
@@ -141,7 +150,10 @@ def has_uav_completed_guided_path(self):
         """
         Returns True if the UAV has completed the guided path, False if not
         """
-        return self.current_path_index >= len(self.current_path)
+        if len(self.current_path) != 0:
+            return self.current_path_index >= len(self.current_path)
+        else:
+            return True
```
Comment (DarkmatterVale) [inline SUASSystem/SUASSystem/sda_converter.py]: In this case, you do not need the else statement. Simply put the ```return True``` after the if statement, and if the condition is not met, ```True``` will still always be returned.

File: SUASSystem/SUASSystem/sda_converter.py [hunk 24]
```diff
@@ -157,6 +169,7 @@ def get_uav_avoid_coordinates(self):
         for xy_loc_point in self.current_path:
             gps_points.append(inverse_haversine(self.initial_coordinates, xy_loc_point))
 
+        #return gps_points
```
Comment (DarkmatterVale) [inline SUASSystem/SUASSystem/sda_converter.py]: Please remove the commented out code if it is not needed.

File: scripts/run_dronekit_sitl.py [hunk 25]
```diff
@@ -2,7 +2,7 @@
 from time import sleep
 
 if __name__ == '__main__':
-    sitl = dronekit_sitl.start_default(lat=38.870571, lon=-77.322357)
+    sitl = dronekit_sitl.start_default(lat=38.870901, lon=-77.322163)
```
Comment (DarkmatterVale) [inline scripts/run_dronekit_sitl.py:5]: Please leave this at FHS.

File: tests/unit_tests/test_obstacle_map.py [hunk 26]
```diff
@@ -157,10 +157,12 @@ def test_generate_possible_path(self):
...
 		self.obstacle_map1.reset_obstacles()
+		self.obstacle_map1.reset_waypoints()
+		self.obstacle_map1.set_drone_position(numpy.array([0,0,0]))
 		self.obstacle_map1.add_waypoint(numpy.array([100, 100, 2]))
-		self.test_obstacle = StationaryObstacle(numpy.array([150, 150, 2]), 10, 300)
-		self.test_waypoint = numpy.array([100, 100, 2])
-		self.test_waypoint2 = numpy.array([600, 800, 700])
+		self.test_obstacle = StationaryObstacle(numpy.array([150, 150, 1]), 10, 300)
+		#self.test_waypoint = numpy.array([100, 100, 2])
+		#self.test_waypoint2 = numpy.array([600, 800, 700])
```
Comment (DarkmatterVale) [inline tests/unit_tests/test_obstacle_map.py:165]: Why did you change the unit tests?

Comment (DarkmatterVale) [reply tests/unit_tests/test_obstacle_map.py:165]: Or more specifically why did you comment out these test waypoints?

Comment (DarkmatterVale) [reply tests/unit_tests/test_obstacle_map.py:165]: If they are not necessary please remove them.

**Review by DarkmatterVale — CHANGES_REQUESTED** [review]
Please note as well the travis build is failing.

File: SDAPackage/SDA/drone.py (hunk 1 above)
Comment (BrightStraightReality) [reply SDAPackage/SDA/drone.py]: done

File: SDAPackage/SDA/obstacle_map.py (hunk 2 above)
Comment (BrightStraightReality) [reply SDAPackage/SDA/obstacle_map.py]: done

File: SDAPackage/SDA/obstacle_map.py (hunk 4 above)
Comment (BrightStraightReality) [reply SDAPackage/SDA/obstacle_map.py]: done

File: SDAPackage/SDA/obstacle_map.py (hunk 3 above)
Comment (BrightStraightReality) [reply SDAPackage/SDA/obstacle_map.py]: done

File: SDAPackage/SDA/obstacle_map.py (hunk 5 above)
Comment (BrightStraightReality) [reply SDAPackage/SDA/obstacle_map.py]: done

File: SDAPackage/SDA/obstacle_map.py (hunk 6 above)
Comment (BrightStraightReality) [reply SDAPackage/SDA/obstacle_map.py]: done

File: SDAPackage/SDA/obstacle_map.py (hunk 7 above)
Comment (BrightStraightReality) [reply SDAPackage/SDA/obstacle_map.py]: done(?)

File: SDAPackage/SDA/obstacle_map.py (hunk 8 above)
Comment (BrightStraightReality) [reply SDAPackage/SDA/obstacle_map.py]: done

File: SUASSystem/SUASSystem/gcs.py (hunk 9 above)
Comment (BrightStraightReality) [reply SUASSystem/SUASSystem/gcs.py]: done

File: SUASSystem/SUASSystem/gcs.py (hunk 10 above)
Comment (BrightStraightReality) [reply SUASSystem/SUASSystem/gcs.py]: done

File: SUASSystem/SUASSystem/gcs.py (hunk 11 above)
Comment (BrightStraightReality) [reply SUASSystem/SUASSystem/gcs.py:67]: done

File: SUASSystem/SUASSystem/sda.py (hunk 12 above)
Comment (BrightStraightReality) [reply SUASSystem/SUASSystem/sda.py]: done

File: SUASSystem/SUASSystem/sda.py (hunk 13 above)
Comment (BrightStraightReality) [reply SUASSystem/SUASSystem/sda.py]: done

File: SUASSystem/SUASSystem/sda.py (hunk 14 above)
Comment (BrightStraightReality) [reply SUASSystem/SUASSystem/sda.py]: done

File: SUASSystem/SUASSystem/sda.py (hunk 15 above)
Comment (BrightStraightReality) [reply SUASSystem/SUASSystem/sda.py]: done

File: SUASSystem/SUASSystem/sda.py (hunk 16 above)
Comment (BrightStraightReality) [reply SUASSystem/SUASSystem/sda.py]: done

File: SUASSystem/SUASSystem/sda.py (hunk 17 above)
Comment (BrightStraightReality) [reply SUASSystem/SUASSystem/sda.py]: done

File: SUASSystem/SUASSystem/sda.py (hunk 18 above)
Comment (BrightStraightReality) [reply SUASSystem/SUASSystem/sda.py:62]: ehhh

File: SUASSystem/SUASSystem/sda_converter.py (hunk 19 above)
Comment (BrightStraightReality) [reply SUASSystem/SUASSystem/sda_converter.py]: lol done

File: SUASSystem/SUASSystem/sda_converter.py (hunk 20 above)
Comment (BrightStraightReality) [reply SUASSystem/SUASSystem/sda_converter.py]: done

File: SUASSystem/SUASSystem/sda_converter.py (hunk 21 above)
Comment (BrightStraightReality) [reply SUASSystem/SUASSystem/sda_converter.py]: done

File: SUASSystem/SUASSystem/sda_converter.py (hunk 22 above)
Comment (BrightStraightReality) [reply SUASSystem/SUASSystem/sda_converter.py]: done

File: SUASSystem/SUASSystem/sda_converter.py (hunk 23 above)
Comment (BrightStraightReality) [reply SUASSystem/SUASSystem/sda_converter.py]: done

File: SUASSystem/SUASSystem/sda_converter.py (hunk 24 above)
Comment (BrightStraightReality) [reply SUASSystem/SUASSystem/sda_converter.py]: done

File: scripts/run_dronekit_sitl.py (hunk 25 above)
Comment (BrightStraightReality) [reply scripts/run_dronekit_sitl.py:5]: (?) talk in class

File: SDAPackage/SDA/obstacle_map.py [hunk 27]
```diff
@@ -216,49 +221,18 @@ def does_path_intersect_obstacle_2d(self, obstacle, uav_point, waypoint):
...
-        print("Distance From Obstacle: " + str(VectorMath.get_vector_magnitude(np.subtract(drone_point, obstacle.get_point()))))
-
-        if self.is_obstacle_in_path_of_drone(obstacle_vector, waypoint_vector):
-            return rejection_vector_magnitude < Constants.STATIONARY_OBSTACLE_SAFETY_RADIUS
-
-        return False
-
     def is_obstacle_in_path_of_drone(self, obstacle_vector, waypoint_vector):
         """
         Looks at the signs of the components of the vectors to determine if the
```
Comment (jvillemare) [inline SDAPackage/SDA/obstacle_map.py:232]: All function docs should define args, type of args, and what it returns (if it returns anything.)

Example:

```
def is_banana(self, fruit):
    """
    Determines whether or not a fruit object is a banana.

    :param fruit:    Object to be evaluated.
    :type fruit:     Fruit object.

    :return:         True/false.
    """
```

See additional example in @Inffzy 's code: https://github.com/FlintHill/SUAS-Competition/blob/master/UpdatedImageProcessing/TargetDetection/automatic_tester.py#L10-L25

Comment (jvillemare) [reply SDAPackage/SDA/obstacle_map.py:232]: @BrightStraightReality you do this in some places, but not everywhere.

File: SUASSystem/SUASSystem/sda.py [hunk 28]
```diff
@@ -2,44 +2,94 @@
...
+            }
+        ]'''
     log(logger_name, "Instantiating SDA converter")
     while True:
         try:
+            print('tries to instantiate')
+            #sda_converter = SUASSystem.SDAConverter(vehicle_state_data[0].get_location(), 5)#mission_information_data[0]["fly_zones"])
             sda_converter = SUASSystem.SDAConverter(vehicle_state_data[0].get_location(), mission_information_data[0]["fly_zones"])
             break
         except:
```
Comment (jvillemare) [inline SUASSystem/SUASSystem/sda.py]: Ideally, you shouldn't use try-catch-alls, but if you have no other workaround, then that's fine.

File: SUASSystem/SUASSystem/sda.py (hunk 14 above)
Comment (jvillemare) [inline SUASSystem/SUASSystem/sda.py]: Commented out code should be removed if possible.

File: SUASSystem/SUASSystem/sda_converter.py [hunk 29]
```diff
@@ -61,11 +60,12 @@ def set_waypoint(self, new_waypoint):
...
-
+        print('waypoint')
+        print(converted_waypoint)
         self.obstacle_map.reset_waypoints()
         self.obstacle_map.add_waypoint(converted_waypoint)
 
-    def add_obstacle(self, obstacle_location, obstacle):
+    def add_obstacle(self, obstacle):
         """
         Add an obstacle to the obstacle map
```
Comment (jvillemare) [inline SUASSystem/SUASSystem/sda_converter.py:69]: In function docs, description and items (`:param obstacles: ...`, `:type obstacle: ...`, `:return: ...`) should end with periods for readability

File: SUASSystem/SUASSystem/gcs.py [hunk 30]
```diff
@@ -48,24 +50,25 @@ def gcs_process(sda_status, img_proc_status, interop_client_array, targets_to_su
...
+                sda_avoid_feet_height = Location(sda_avoid_coords[0].get_lat(), sda_avoid_coords[0].get_lon(), sda_avoid_coords[0].get_alt()*3.28084)
+                print('sda avoid feet height')
+                print(sda_avoid_feet_height)
                 log("root", "Avoiding obstacles...")
-                vehicle.mode = VehicleMode("GUIDED")
-                guided_waypoint_location = sda_avoid_coords[0][0]
-                vehicle.simple_goto(guided_waypoint_location.as_global_relative_frame())
+                vehicle.mode = dronekit.VehicleMode("GUIDED")
+                vehicle.simple_goto(sda_avoid_coords[0].as_global_relative_frame())
+            if UAV_status.value == 'AUTO' and vehicle.mode.name != "AUTO":
```
Comment (jvillemare) [inline SUASSystem/SUASSystem/gcs.py:66]: There should be a new line above a new if statements. You do this in line 57 of this file, but not here.

**Review by jvillemare — CHANGES_REQUESTED** [review]
Overall, formatting is great. Some docs just need to be expanded/cleaned up, and some new lines here and there for readability.

**Comment by DarkmatterVale** [comment]
I'm going to go ahead and merge this code in. The formatting errors are mostly fixed, and the rest do not look that important.
